src/buffer.js

```diff
@@ -0,0 +1,127 @@
+const {contextRecordKey, deferred} = require('./utils');
+
+function queryBuffer(config, emitter, targetStore) {
+  const buffers = [];
+
+  let numCoalesced = 0;
+  let numCached = 0;
+  let numMisses = 0;
+
+  class RequestBuffer {
+    constructor(key, params) {
+      this.uid = Math.random().toString(36);
+      this.state = 0;
+      this.ids = [];
+      this.contextKey = key;
+      this.params = params;
+      this.contexts = [];
+      this.handle = deferred();
+      this.timer = null;
+    }
+
+    tick() {
+      const sizeLimit = config.batch?.limit || 1;
```

Comment:
[🧗] Do you think evaluating those values only once could give a performance benefit?

Reply:
If it's microscopic please disregard, let's not impair the readability of the code for that.

Reply:
Indeed, not a huge gain.